tests/runner/src/main.rs

```diff
@@ -45,61 +49,103 @@ fn main() {
+
+        // Check on test requirements before running the test
+        if !FluvioTest::is_env_acceptable(
+            &(test.requirements)(),
+            &TestCase::new(option.environment.clone(), test_opt.clone()),
+        ) {
+            exit(-1);
+        }
 
         let panic_timer = TestTimer::start();
-
-        let test_result = panic::catch_unwind(AssertUnwindSafe(move || {
-            let test = inventory::iter::<FluvioTest>
-                .into_iter()
-                .find(|t| t.name == test_name.as_str())
-                .expect("Test not found");
-
-            // Run the test
-            (test.test_fn)(fluvio_client.clone(), test_case)
-        }));
-
-        // Handle panics from user tests
         std::panic::set_hook(Box::new(move |_panic_info| {
-            let mut inside_timer = panic_timer.clone();
-            inside_timer.stop();
+            let mut panic_timer = panic_timer.clone();
+            panic_timer.stop();
 
             let test_result = TestResult {
                 success: false,
-                duration: inside_timer.duration(),
+                duration: panic_timer.duration(),
             };
             //run_block_on(async { cluster_cleanup(panic_options.clone()).await });
             eprintln!("Test panicked:\n");
             eprintln!("{}", test_result);
         }));
 
-        // Cluster cleanup
-        cluster_cleanup(option.environment).await;
-
-        match test_result {
-            // Pass/Fail results
-            Ok(results) => {
-                // We always return TestResults. This is awkward.
-                match results {
-                    Ok(r) => println!("{}", r),
-                    Err(r) => eprintln!("{}", r),
-                }
-            }
-            // Panic results
-            Err(fail_results) => {
-                let test_result = fail_results
-                    .downcast_ref::<TestResult>()
-                    .expect("Failed to convert error from panic")
-                    .to_owned();
-
-                eprintln!("{}", test_result);
-                std::process::exit(-1);
-            }
+        if option.environment.benchmark {
+            run_benchmark(test, fluvio_client, option.environment.clone(), test_opt)
+        } else {
+            let test_result =
+                run_test(option.environment.clone(), test_opt, test, fluvio_client).await;
+            cluster_cleanup(option.environment).await;
+
+            println!("{}", test_result)
         }
     });
 }
 
+async fn run_test(
+    environment: EnvironmentSetup,
+    test_opt: Box<dyn TestOption>,
+    test: &FluvioTest,
+    fluvio_client: Arc<Fluvio>,
+) -> TestResult {
+    let test_case = TestCase::new(environment, test_opt);
+    let test_result = panic::catch_unwind(AssertUnwindSafe(move || {
+        (test.test_fn)(fluvio_client.clone(), test_case)
+    }))
+    .expect("Panic hook should have caught this");
+
+    test_result.expect("Test Result")
```

Comment:
What is the behavior here? It seems like you're catching panics, but then just unwrapping and panicking anyway? Is this any different than not catching the panic at all?